crates/core/component/dex/src/component/mod.rs

```diff
@@ -11,15 +11,16 @@ mod arb;
 pub(crate) mod circuit_breaker;
 mod dex;
 mod flow;
-pub(crate) mod position_counter;
-pub(crate) mod position_manager;
+mod position_manager;
 mod swap_manager;
 
 pub use self::metrics::register_metrics;
-pub use arb::Arbitrage;
+pub(crate) use arb::Arbitrage;
 pub use circuit_breaker::ExecutionCircuitBreaker;
 pub(crate) use circuit_breaker::ValueCircuitBreaker;
 pub use dex::{Dex, StateReadExt, StateWriteExt};
+// TODO(erwan): exposing a DEX interface to other components
+// is useful but maybe we should restrict it to open/queue/close positions
```

Comment:
Isn't that what the `PositionManager` is?

Reply:
yes, i was think we might want to restrict it further for example wrt closing positions, and force consumers to use the queue

Reply:
Ah, well, the auction component as designed doesn’t allow that. So i would guess we probably wouldn’t